src/hooks/use-debounced.ts

```diff
@@ -1,11 +1,20 @@
-import { useState, useEffect } from 'react';
+import { useState, useEffect, useRef } from 'react';
 
 export default function useDebounced<T>(value: T, delay: number) {
+  const isInitialMount = useRef(true);
   const [debounced, setDebounced] = useState(value);
 
+  // eslint-disable-next-line consistent-return
   useEffect(() => {
-    const timeout = setTimeout(() => setDebounced(value), delay);
-    return () => clearTimeout(timeout);
+    if (isInitialMount.current) {
+      isInitialMount.current = false;
+    } else {
+      const timeout = setTimeout(() => {
+        setDebounced(value);
+      }, delay);
+
+      return () => clearTimeout(timeout);
+    }
```

Comment:
I think this will alleviate the need for the eslint disable comment...

```suggestion
    }
    
    return () => {};
```

Reply:
Interesting. I had that but I thought "Lars is going to yell at me for not having a comment on this awkward noop line."

Reply:
Haha yeah it's an odd one. Always nice to avoid disabling eslint when possible though.